src/game/Utils/Event_Pump.cc

```diff
@@ -150,15 +206,28 @@ static SOLDIERTYPE* GetSoldier(const UINT16 soldier_idx)
 }
 
 
-static BOOLEAN ExecuteGameEvent(EVENT* pEvent)
+// Copied from https://stackoverflow.com/posts/52303671/revisions
+// Author: Barry
+template<typename VariantType, typename T, std::size_t index = 0>
+    constexpr std::size_t variant_index() {
+        if constexpr (index == std::variant_size_v<VariantType>) {
+            return index;
+        } else if constexpr (std::is_same_v<std::variant_alternative_t<index, VariantType>, T>) {
+            return index;
+		} else {
```

Comment:
Can you remove the extra indent to better align the code?

Reply:
Or alternatively, would you consider using `std::visit` instead? So it is more idiomatic and will look like:

```c++
    std::visit(overload{
        [](EV_S_GETNEWPATH& enp) { /* do something */ },
        [](EV_S_SETDESIREDDIRECTION& edd) {  /* do something else */ },
        // and so on...
    }, pEvent->gameEvent);
```

Reply:
Another avid cppreference reader, I see :+1:. Sure, I had at first decided against it because it wanted to make it easier to verify that the actual logic in ExecuteGameEvent is unchanged, but overall I like the std::visit solution more, too.